pkg/monitor/worker.go

```diff
@@ -18,6 +18,23 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/restconfig"
 )
 
+// This function will recurse until such time as it has a config to add to the global Hive shard map
+// Note that because the mon.hiveShardConfigs[shard] is set to `nil` when its created the cluster
+// monitors will simply ignore Hive stats until this function populates the config
+func (mon *monitor) populateHiveShardRestConfig(ctx context.Context, shard int) {
+	hiveRestConfig, err := mon.liveConfig.HiveRestConfig(ctx, shard)
+	if err != nil {
+		mon.baseLog.Printf("error fetching Hive kubeconfig for shard %d: %s", shard, err.Error())
+		mon.baseLog.Printf("pausing for a minute before retrying...")
+		time.Sleep(time.Duration(60 * time.Second))
+		mon.populateHiveShardRestConfig(ctx, shard)
```

Comment:
For permanent errors like our current 403 we might run out of memory over time with one recursive call per minute.  Would it make sense to change this to a loop ?

Reply:
Agree with @UlrichSchlueter , that or having a different context than the one from the changefeed so there is a way out.

Reply:
I refactored this not to use recursion.